dali/dfu/dfuwu.cpp

```diff
@@ -2505,8 +2567,10 @@ class CDFUWorkUnit: public CDFUWorkUnitBase
             case DFUstate_finished:
                 return ret;
             }
-            if (!completed.wait(timeout))       // should only go round loop once
-                break;
+            MilliSleep(5000);
+            // TBD figure out why subsctiption + Semaphore isn't working
```

Comment:
In what way isn't it working?

Reply:
When the workunit is changed by dfu server the subscribe isn't being notified.  I've tried a few variations, but this code isn't used anywhere.  All other examples I've found in use poll for workunit completion. I'll do the same, and perhaps this mechanism can be fixed in another PR.